Tests/AK/TestCharacterTypes.cpp

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2018-2020, Andreas Kling <kling@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibTest/TestCase.h>
+
+#include <AK/CharacterTypes.h>
+#include <AK/NumericLimits.h>
+#include <ctype.h>
+
+#define ASCII 0x80
+#define UNICODE 0x10FFFF + 100
+
+#define COMPARE_BOOLEAN_OUTPUT_OVER(range, func1, func2)               \
+    bool result1 = false;                                              \
+    bool result2 = false;                                              \
+    for (u32 i = 0; i < range; ++i) {                                  \
+        EXPECT_EQ(result1 = (func1(i) > 0), result2 = (func2(i) > 0)); \
+        if (result1 != result2)                                        \
+            dbgln("Function input value was {}.", i);                  \
+    }
+
+#define COMPARE_VALUE_OUTPUT_OVER(range, func1, func2)     \
```

Comment:
These can (and probably should) just be real functions, no need for them to be macros as far as I can tell?

Reply:
I'm not saying get rid of the common validation, I'm saying it doesn't need to be implemented as a macro. It can be a real function.

Reply:
I didn't think about that, somehow though this wasn't even possible. Sorry!